standards/nextjs_standards.md

```diff
@@ -0,0 +1,95 @@
+- Adopt a consistent indentation method (e.g., 2 spaces) across the project.
+
+## Naming Conventions
+
+- **Files**: Use PascalCase for files (e.g., `MyComponent.ts`).
+- **Directories**: Use kebab-case for directories (e.g., `my-directory`).
+- **Components**: Use PascalCase for React components (e.g., `UserProfile.js`).
+- **Variables and Functions**: Use camelCase for variables and functions (e.g., `getUserData`).
+
+## File Structure & Naming Convention
+
+- Organize project files under the `src` directory for clarity.
+- Use the `pages` directory for page components, leveraging Next.js's file-based routing.
+- Store static assets in the `public` directory.
+- Group components into a `components` directory, categorizing further if necessary.
+
+## Component Structure
+
+Use atomic design when organizing components. Organize components into atoms, molecules, organisms, and templates for scalability and maintainability.
+
+## Efficient Code Splitting, Dynamic Imports, and Loading of Client-Side Scripts
+
+- Leverage Next.js's built-in code splitting to manage application code in smaller, more manageable units.
+- Utilize dynamic imports for loading non-essential components on demand, reducing the initial bundle size and improving page load times.
+- Identify components or code segments not critical for the initial page load and load them asynchronously.
+
+## Rendering Modes
+
+- **Server-Side Rendering (SSR)**: Use for pages requiring dynamic content or SEO, ideal for public-facing and content-heavy pages.
+- **Static Site Generation (SSG)**: Use for content with infrequent changes, suitable for landing pages, blogs, and marketing sites.
+- **Client-Side Rendering (CSR)**: Use for interactive components and real-time data, prioritizing pages/components that require interactivity.
+- **Hybrid Rendering**: Combine SSR and CSR benefits for applications with mixed content and interactivity needs.
+
+## Styling
+
+- Utilize CSS Modules for component-level styles to avoid conflicts.
+- Consider CSS-in-JS solutions like styled-components for dynamic styling.
+- Use the `Image` component from `next/image` for optimized image handling.
+- Use utility-first CSS frameworks like Tailwind CSS.
+
+## Accessibility
+
+- Follow Web Content Accessibility Guidelines (WCAG) to ensure your application is accessible.
+- Use semantic HTML and ARIA attributes where necessary.
+- Test accessibility with tools and aim for keyboard navigability and screen reader compatibility.
+
+## Performance Optimization
+
+- Optimize images and leverage Next.js's image optimization.
+- Use static generation (SSG) where possible for faster page loads.
+- Analyze and optimize bundle sizes with Next.js's built-in analytics.
+
+## Security Practices
+
+- Sanitize user input to prevent XSS attacks.
+- Use environment variables for sensitive information.
+- Regularly update dependencies to mitigate vulnerabilities.
```

Comment:
Add samples like for sanitize, let's add our preferred our way and library to do this

Reply:
Got this 👌🏼